extensions/azure/src/dependabot/branch-name.ts

```diff
@@ -28,17 +28,27 @@ export function getBranchNameForUpdate(
     const dependencyNames = dependencies
       .map((d) => d['dependency-name'])
       .join('-and-')
-      .replace(/[:\[\]]/g, '-') // Replace `:` and `[]` with `-`
+      .replace(/[:[]]/g, '-') // Replace `:` and `[]` with `-`
```

Comment:
@mburumaxwell does this need a test? 
Would `dependabot/something[14435324]-1.0` be sanitised to `dependabot/something-14435324--1.0` after this change? My understanding is that those escape are nessecary otherwise regex will treat it as an empty "any of" condition.

I have no idea what/why `[` and `]` can appear in branch names, but this was checked in dependabot-core:
https://github.com/dependabot/dependabot-core/blob/3569031a5c513cac6c57a908619dd8c0f95a0270/common/lib/dependabot/pull_request_creator/branch_namer/solo_strategy.rb#L32